types/transaction.go

```diff
@@ -1,12 +1,45 @@
 package types
 
 import (
+	"fmt"
 	"math/big"
 	"sync/atomic"
 
 	"github.com/0xPolygon/polygon-edge/helper/keccak"
 )
 
+// TransactionType represents the value between 0x00 and 0x7F to represents the type of transaction
+// TransactionType is put before transaction data in RLP encoded
+type TransactionType uint8
+
+const (
+	TxTypeLegacy TransactionType = 0x0
+	// TxTypeAccessList TransactionType = 0x1
+	// TxTypeDynamicFee TransactionType = 0x2
+	TxTypeState TransactionType = 0x7f
+)
+
+func ToTransactionType(b byte) (TransactionType, error) {
+	tt := TransactionType(b)
+	switch tt {
+	case TxTypeLegacy, TxTypeState:
+		return tt, nil
+	default:
+		return TransactionType(0), fmt.Errorf("undefined transaction type: %d", b)
+	}
+}
+
+func (t TransactionType) String() string {
+	switch t {
+	case TxTypeLegacy:
+		return "TxTypeLegacy"
+	case TxTypeState:
+		return "TxTypeState"
+	default:
+		return fmt.Sprintf("TxType(%x)", byte(t))
+	}
+}
```

Comment:
Why support just these 2?

Reply:
Same to https://github.com/0xPolygon/polygon-edge/pull/434#discussion_r825895761